Panic in `GetKeyVaulSecret`

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1956479&view=logs&j=38b6934a-871b-50c9-78fb-2dab4cf5768f&t=c4f60ece-0f0e-5c13-176a-be73979c75d1

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x832185]

goroutine 26 [running]:
github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets.(*ID).Name(...)
	/home/cloudtest/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/keyvault/azsecrets@v0.10.1/custom_client.go:50
github.com/azure/azure-dev/cli/azd/pkg/tools/azcli.(*azCli).GetKeyVaultSecret(0x20?, {0xbfd878, 0xc0000b84e0}, {0xc0000f66c0, 0x8}, {0xc0000f66c9, 0x10})
	/mnt/vss/_work/1/s/cli/azd/pkg/tools/azcli/keyvault.go:88 +0x245
github.com/azure/azure-dev/cli/azd/pkg/cmdsubst.(*SecretOrRandomPasswordCommandExecutor).Run(0xc000098230, {0xbfd878, 0xc0000b84e0}, {0xc0000f66a9?, 0xc0000f0000?}, {0xc0000ba360, 0x2, 0x0?})
	/mnt/vss/_work/1/s/cli/azd/pkg/cmdsubst/secretOrRandomPassword.go:56 +0x10f
github.com/azure/azure-dev/cli/azd/pkg/cmdsubst.Eval({0xbfd878, 0xc0000b84e0}, {0xc0000f6500, 0x24b}, {0xbf7c00, 0xc000098230})
	/mnt/vss/_work/1/s/cli/azd/pkg/cmdsubst/cmdsubst.go:67 +0x365
github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning/bicep.(*BicepProvider).createParametersFile(0xc0004ba080, {0xbfd878, 0xc0000b84e0}, 0x0?)
	/mnt/vss/_work/1/s/cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go:775 +0x28e
github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning/bicep.(*BicepProvider).Plan.func1(0xc0000b8510)
	/mnt/vss/_work/1/s/cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go:138 +0x105
github.com/azure/azure-dev/cli/azd/pkg/async.(*InteractiveTaskWithProgress[...]).Run.func1()
	/mnt/vss/_work/1/s/cli/azd/pkg/async/task.go:191 +0x1a8
created by github.com/azure/azure-dev/cli/azd/pkg/async.(*InteractiveTaskWithProgress[...]).Run
	/mnt/vss/_work/1/s/cli/azd/pkg/async/task.go:186 +0x1cb
```

--- a/cli/azd/pkg/tools/azcli/keyvault.go
+++ b/cli/azd/pkg/tools/azcli/keyvault.go
@@ -73,14 +73,10 @@ func (cli *azCli) GetKeyVaultSecret(ctx context.Context, vaultName string, secre
 	response, err := client.GetSecret(ctx, secretName, "", nil)
 	if err != nil {
 		var httpErr *azcore.ResponseError
-
-		if errors.As(err, &httpErr) {
-			if httpErr.StatusCode == http.StatusNotFound {
-				return nil, ErrAzCliSecretNotFound
-			}
-		} else {
-			return nil, fmt.Errorf("getting key vault secret: %w", err)
+		if errors.As(err, &httpErr) && httpErr.StatusCode == http.StatusNotFound {
+			return nil, ErrAzCliSecretNotFound
 		}
+		return nil, fmt.Errorf("getting key vault secret: %w", err)
 	}
 
 	return &AzCliKeyVaultSecret{

